study1_200204/kyungyeon.js

```diff
@@ -0,0 +1,61 @@
+const answer1 = (n, sum = 1) => {
+  if (n === 0) {
+    return sum;
+  }
+
+  sum = n * sum;
+  return answer1(n-1, sum);
+};
+
+const answer2 = arrSize => {
+  const arr = [[], []];
+  for (let i = 0; i < arrSize; i++) {
+    arr[i] = new Array(arrSize).fill(0);
+  }
+
+  let direction = 1;
+  let col = 0, row = -1; // 최초 시작값을 0, 0 맞추기 위해 row는 -1에서 시작
+  let maxLimitCount = arrSize;
+  let num = 1;
+
+  // set data
+  while(true) {
+    // 달팽이를 돌다가 더이상 돌 공간이 없다면 종료
+    if (maxLimitCount === 0) {
+      break;
+    }
```

Comment:
`while` 문 안에 `if` 문이 별로도 존재하는 이유가 있을까요? 
```suggestion
    while(maxLimitCount > 0) {
 
    (line 35. maxLimitCount--;)
```
위 방식과 다른점은 어떤게 더 있을까요? (성능이라던가..? 혹은 메모리 참조라던가.. 이득이 있나요?)